agenta-backend/agenta_backend/services/db_manager.py

```diff
@@ -326,24 +330,40 @@ async def remove_app_variant(app_variant: AppVariant, **kwargs: dict):
         & query.eq(AppVariantDB.user_id, user.id)
     )
 
+    # Build the query expression to delete variants with is_deleted flag
+    delete_var_query_expression = (
+        query.eq(AppVariantDB.app_name, app_variant.app_name)
+        & query.eq(AppVariantDB.user_id, user.id)
+        & query.eq(AppVariantDB.is_deleted, True)
+    )
+
     # Get app variant
     app_variant_db = await engine.find_one(AppVariantDB, query_expression)
+
+    # Get variant with is_deleted flag
+    pending_variant_to_delete = await engine.find_one(
+        AppVariantDB, delete_var_query_expression
+    )
     is_last_variant = await check_is_last_variant(app_variant_db)
     if app_variant_db is None:
         raise ValueError("App variant not found")
 
     if app_variant_db.previous_variant_name is not None:  # forked variant
         await engine.delete(app_variant_db)
+        if pending_variant_to_delete is not None:
+            await engine.delete(pending_variant_to_delete)
 
     elif is_last_variant:  # last variant using the image, okay to delete
         await engine.delete(app_variant_db)
+        if pending_variant_to_delete is not None:
+            await engine.delete(pending_variant_to_delete)
```

Comment:
The second cause of the bug, as mentioned in the PR description, is as follows: Without the condition that checks for a pending_variant_to_delete, the following scenario occurs: after an app has been deleted, its first associated variant remains unaffected. Although the flag `is_deleted` is set to True, I discovered this issue after extensive debugging.

With the added changes, the process is now as follows: when the condition checks whether the current variant to be deleted is the last variant (which, by default, has `is_deleted` set to False), and if this condition is met, the system then proceeds to check whether there exists a variant that is yet to be deleted.

Does my explanation make sense now?